test/composableStableJoin.integration.test.ts

```diff
@@ -0,0 +1,311 @@
+        );
+    });
+
+    test('native asset join', async () => {
+        const bptIndex = txInput.poolInput.tokens.findIndex(
+            (t) => t.address === txInput.poolInput.address,
+        );
+
+        const poolTokensWithoutBpt = txInput.poolInput.tokens
+            .map((t) => new Token(chainId, t.address, t.decimals))
+            .filter((_, index) => index !== bptIndex);
+
+        const amountsIn = poolTokensWithoutBpt.map((t) =>
+            TokenAmount.fromHumanAmount(t, '1'),
+        );
+
+        // perform join query to get expected bpt out
+        const joinInput: UnbalancedJoinInput = {
+            amountsIn,
+            chainId,
+            rpcUrl,
+            kind: JoinKind.Unbalanced,
+            useNativeAssetAsWrappedAmountIn: true,
+        };
+
+        // We have to use zero address for balanceDeltas
+        const { queryResult, maxAmountsIn, minBptOut, value } = await doJoin({
+            ...txInput,
+            joinInput,
+            checkNativeBalance: true,
+        });
+        // Query should use same amountsIn as user sets
+        expect(
+            queryResult.amountsIn
+                .filter((_, index) => index !== bptIndex)
+                .map((a) => a.amount),
+        ).to.deep.eq(amountsIn.map((a) => a.amount));
+        expect(queryResult.tokenInIndex).to.be.undefined;
+        // Should have native value equal to input amount
+        expect(value).eq(amountsIn[1].amount);
+
+        // Expect some bpt amount
+        expect(queryResult.bptOut.amount > BigInt(0)).to.be.true;
+
+        // Confirm slippage - only bpt out
+        const expectedMinBpt = txInput.slippage.removeFrom(
+            queryResult.bptOut.amount,
+        );
+        expect(expectedMinBpt).to.deep.eq(minBptOut);
+        const expectedMaxAmountsIn = amountsIn.map((a) => a.amount);
+        expect(expectedMaxAmountsIn).to.deep.eq(
+            maxAmountsIn.filter((_, index) => index !== bptIndex),
+        );
+    });
+
+    test('single asset join', async () => {
+        const bptIndex = txInput.poolInput.tokens.findIndex(
+            (t) => t.address === txInput.poolInput.address,
+        );
+        const bptOut = TokenAmount.fromHumanAmount(bptToken, '1');
+        const tokenIn = '0x4bc3263eb5bb2ef7ad9ab6fb68be80e43b43801f';
+
+        // perform join query to get expected bpt out
+        const joinInput: SingleAssetJoinInput = {
+            bptOut,
+            tokenIn,
+            chainId,
+            rpcUrl,
+            kind: JoinKind.SingleAsset,
+        };
+
+        const { queryResult, maxAmountsIn, minBptOut, value } = await doJoin({
+            ...txInput,
+            joinInput,
+        });
+        // Query should use same bpt out as user sets
+        expect(queryResult.bptOut.amount).to.deep.eq(bptOut.amount);
+
+        // We only expect single asset to have a value for amount in
+        expect(queryResult.tokenInIndex).toBeDefined;
```

Comment:
Because of the issue we saw before testing against undefined lets use:
`expect(checkedValue === undefined).toEqual(true);`